src/utilities/util_dump.c

```diff
@@ -210,13 +220,26 @@ util_dump(WT_SESSION *session, int argc, char *argv[])
 
         if (dump_config(session, simpleuri, cursor, pretty, hex, json) != 0)
             goto err;
-        if (key == NULL) {
+
+        if(key == NULL) {
+            if (start_key != NULL) {
+                cursor->set_key(cursor, start_key);
+                if (cursor->bound(cursor, "action=set,bound=lower") != 0)
+                    goto err;
+            }
+            if (end_key != NULL) {
+                cursor->set_key(cursor, end_key);
+                if (cursor->bound(cursor, "action=set,bound=upper") != 0)
+                    goto err;
+            }
             if (dump_all_records(cursor, reverse, json) != 0)
                 goto err;
-        } else {
-            if (dump_record(cursor, key, reverse, search_near, json, window) != 0)
+            if ((start_key != NULL || end_key != NULL) &&
+              cursor->bound(cursor, "action=clear") != 0)
```

Comment:
If only the `start_key` or `end_key` are bound above does "action=clear" still need to be called?

Reply:
Yep!

Reply:
Never mind completely misread that as `&&` ?!?